projects/plugins/social/src/class-jetpack-social.php

```diff
@@ -130,4 +136,26 @@ public function plugin_settings_page() {
 			<div id="jetpack-social-root"></div>
 		<?php
 	}
+
+	/**
+	 * Activate the Publicize module on plugin activation.
+	 *
+	 * @static
+	 */
+	public static function plugin_activation() {
+		add_option( self::JETPACK_SOCIAL_ACTIVATION_OPTION, true );
+	}
+
+	/**
+	 * Runs an admin_init and checks the activation option to work out
+	 * if we should activate the module. This needs to be run after the
+	 * activation hook, as that results in a redirect, and we need the
+	 * sync module's actions and filters to be registered.
+	 */
+	public function activate_module_on_plugin_activation() {
+		if ( get_option( self::JETPACK_SOCIAL_ACTIVATION_OPTION ) ) {
+			delete_option( self::JETPACK_SOCIAL_ACTIVATION_OPTION );
```

Comment:
Is the intent behind `delete_option( self::JETPACK_SOCIAL_ACTIVATION_OPTION );` is to avoid activation, if the module is already active? 

If yes, can we do this?
```
$is_active = ( new Modules() )->is_active( Jetpack_Social::JETPACK_PUBLICIZE_MODULE_SLUG ) 
if ( ! $is_active ) {
   ( new Modules() )->activate( self::JETPACK_PUBLICIZE_MODULE_SLUG, false, false );
}

Reply:
Not exactly. The code for syncing etc. Isn't loaded as the plugin is first activated, so we set an option that we then check on the next request that runs `admin_init`. If the option is there we remove it so that we don't keep trying to activate the plugin.

Reply:
Got you, thanks!